### DISCUSSION TITLE
Desktop app

### DISCUSSION THREAD

**Discussion by gitToad** [eidolonFIRE/xcNav_web-DEPRECATED-#13] [Ideas]
In addition to the phone based flying optimized one-finger UI client app, we also plan to have a desktop app.

Probably built around the same UI components it is aimed at evaluating and managing the pilot and flight telemetry data we produce. It takes advantage of much larger screen, more memory and CPU, and mouse based interactions.

It would likely be able to:

- replay meetups with a time slider
- replay would include pilots on the map, their paths, but also in-flight generated data such as messages, shared markers, take off and landing events etc.
- cumulative statistics would be easy to gather per meetup or per pilot over several meetups
- exporting data for other analysis in useful formats (CSV etc.)
- managing data (for development it can be very useful to have browse table, insert row, delete row, edit row capabilities into all DB tables (like phpmyadmin but less clumsy)
- ability to dump the complete database into a .sql text file is useful for making snapshots for development or testing purposes as well as to make periodic backups in case something bad happens or we want to move servers

**Comment by eidolonFIRE** [comment]
I definitely agree there's lots of value in having a desktop interface for doing post analysis. I love this stuff.
I'd love to incorporate some of the work I've already done with post processing the gps tracks. https://github.com/eidolonFIRE/gps_tools

That being said, I feel for now we should be careful about the scope of the project. I think we should first focus on making the phone app. I see the desktop as having a fundamentally different UI so I don't think there's much cross-over between the desktop use-case and the phone case. Phone is for navigation and desktop is for debrief. Very different UI elements will be present. So, I see desktop app as a considerable addition of project scope. IMHO, we should save this for a little further down the road avoid having it influence the project design.
